napari/_qt/widgets/qt_keyboard_settings.py

```diff
@@ -272,11 +300,24 @@ def _get_layer_actions(self):
             viewer_actions = self.key_bindings_strs[self.VIEWER_KEYBINDINGS]
 
             actions_all.update(viewer_actions)
+
+        if current_layer_text is not self.PLUGINS_KEYBINDINGS:
+            plugins_actions = self.key_bindings_strs[
+                self.PLUGINS_KEYBINDINGS
+            ]
+
+            actions_all.update(plugins_actions)
+
         return actions_all
 
     def _restore_shortcuts(self, row):
         action_name = self._table.item(row, self._action_col).text()
-        shortcuts = action_manager._shortcuts.get(action_name, [])
+
+        if action_name in self._app.commands._commands:
+            shortcuts = [str(keybinding.keybinding) for keybinding in self._app.keybindings._keybindings
```

Comment:
the parent `keybindings` object is iterable

```suggestion
            shortcuts = [str(keybinding.keybinding) for keybinding in self._app.keybindings
```